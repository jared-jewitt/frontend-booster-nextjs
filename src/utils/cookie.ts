interface Options {
  path?: string;
  domain?: string;
  secure?: boolean;
  days?: number;
}

export const cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: Options = {}): void {
    if (options.days) {
      options["max-age"] = options.days * 60 * 60 * 24;
      delete options.days;
    }

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string): void {
    this.set(name, "", { days: -1 });
  },
};
