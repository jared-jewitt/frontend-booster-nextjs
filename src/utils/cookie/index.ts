interface IOptions {
  path?: string;
  domain?: string;
  secure?: boolean;
  days: number;
}

export const Cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: IOptions): void {
    options["max-age"] = options.days * 60 * 60 * 24;
    delete options.days;

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string, options?: Pick<IOptions, "path" | "domain">): void {
    this.set(name, "", { ...options, days: -1 });
  },
};
