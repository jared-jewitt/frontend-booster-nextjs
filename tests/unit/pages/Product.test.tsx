import React from "react";
import Product from "@/pages/products/[id]";
import { render } from "../../utils";

describe("Product page", () => {
  it("matches snapshot", () => {
    const mockProduct = {
      id: 123,
      title: "Test product",
      description: "Foo bar baz",
      price: 99.99,
    };

    const { asFragment } = render(<Product product={mockProduct} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
