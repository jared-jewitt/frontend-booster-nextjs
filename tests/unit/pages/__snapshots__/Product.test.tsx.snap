// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <a
        href="/login"
      >
        Login
      </a>
    </div>
    <main
      class="main"
    >
      <div
        class="product"
      >
        <ul>
          <li>
            ID: 123
          </li>
          <li>
            Title: Test product
          </li>
          <li>
            Description: Foo bar baz
          </li>
          <li>
            Price: $99.99
          </li>
        </ul>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
